api/src/test/java/org/apache/iceberg/transforms/TestBucketing.java

```diff
@@ -165,6 +212,57 @@ public void testLong() {
         .isEqualTo(hashBytes(buffer.array()));
   }
 
+  @Test
+  public void testTimestampNanoPromotion() {
```

Comment:
Can we break this test down further? Feel like it's testing a lot in a single method. At least separation between timestamp with and without timezone seems feasible.

Reply:
done